engine/api/src/main/java/org/apache/cloudstack/engine/orchestration/service/VolumeOrchestrationService.java

```diff
@@ -127,4 +129,24 @@ DiskProfile allocateTemplatedVolume(Type type, String name, DiskOffering offerin
     StoragePool findStoragePool(DiskProfile dskCh, DataCenter dc, Pod pod, Long clusterId, Long hostId, VirtualMachine vm, Set<StoragePool> avoid);
 
     void updateVolumeDiskChain(long volumeId, String path, String chainInfo);
+
+    /**
+     * Imports an existing volume for a VM into database. Useful while ingesting an unmanaged VM.
+     * @param type Type of the volume - ROOT, DATADISK, etc
+     * @param name Name of the volume
+     * @param offering DiskOffering for the volume
+     * @param size DiskOffering for the volume
+     * @param minIops minimum IOPS for the disk, if not passed DiskOffering value will be used
+     * @param maxIops maximum IOPS for the disk, if not passed DiskOffering value will be used
+     * @param vm VirtualMachine this volume is attached to
+     * @param template Template of the VM of the volume
```

Comment:
Import of VM(and its disk) is allowed without using any template. But since CS needs a template or ISO ID for a VM, we need to create a dummy, removed ISO which can be used for import in such cases. Will try to add javadoc for this somewhere.